gpytorch/priors/lkj_prior.py

```diff
@@ -107,7 +162,10 @@ def __init__(self, n, eta, sd_prior, validate_args=False):
             raise ValueError("sd_prior must be an instance of Prior")
         if not isinstance(n, int):
             raise ValueError("n must be an integer")
-        if sd_prior.event_shape not in {torch.Size([1]), torch.Size([n])}:
+        # bug-fix event shapes if necessary
```

Comment:
Is this basically a hack around #1317/#1318?

Reply:
No, this is actually a hack around a difference in event shapes between exponential family distributions in `torch.distributions` and the event shape in `SmoothedBoxPrior`. I think it's a bug in torch.distributions.

Reply:
Here's the distinction:

```python
from gpytorch.priors import GammaPrior, SmoothedBoxPrior
prior1 = SmoothedBoxPrior(0.1, 1.0)
prior1.event_shape # torch.Size([1])

prior2 = GammaPrior(3.0, 5.0)
prior2.event_shape # torch.Size([])
```

Reply:
I see. Yeah so I believe all of `torch.distributions` will have an empty event shape if initialized with scalars. So I'm wondering if we should change `SmoothedBoxPrior` to do the same.